hk_adc/hk_adc.c

```diff
@@ -0,0 +1,504 @@
+                // are 12 most significant bits
+                read_value = read_value >> 4;
+                *citi_temp = read_value;
+                return err;
+            }
+        }
+    }
+
+    err = HK_ADC_ERR_TEMP_READ_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC register write function
+ *
+ * This function performs register write operation. Usually, the value to be
+ * written is a 16-bit value as all the registers accessible for a write
+ * operation in HK ADC are 16-bit registers.
+ *
+ * @param reg[in]              Variable of enumeration type #hk_adc_register_t
+ *                             (defined in hk_adc.h) indicating which register
+ *                             is to be written.
+ * @param write_buffer[in]     pointer to an unsigned byte array containing data
+ *                             to be written.
+ * @return HK_ADC_ERR_WRITE_FAILED if any error occurs otherwise HK_ADC_NO_ERR.
+ *
+ */
+int hk_adc_reg_write(hk_adc_register_t reg, uint8_t *write_buffer)
+{
+    uint8_t send_buffer[3] = {0xff};
+    uint8_t err = HK_ADC_ERR_WRITE_FAILED;
+
+    // assign register address pointer
+    switch(reg)
+    {
+        case HK_ADC_REG_CONVERSION:
+            send_buffer[0] = HK_ADC_REG_PTR_CONVERSION;
+            break;
+
+        case HK_ADC_REG_CONFIG:
+            send_buffer[0] = HK_ADC_REG_PTR_CONFIG;
+            break;
+
+        case HK_ADC_REG_LO_THRESH:
+            send_buffer[0] = HK_ADC_REG_PTR_LOW_THRESH;
+            break;
+
+        case HK_ADC_REG_HI_THRESH:
+            send_buffer[0] = HK_ADC_REG_PTR_HI_THRESH;
+            break;
+
+//        default:
+            //no other register possible
+    }
+
+    send_buffer[1] = *write_buffer;         //MSB
+    send_buffer[2] = *(write_buffer + 1);   //LSB
+
+    // Write data to slave.
+    MSS_I2C_write(&g_mss_i2c0, HK_ADC_ADDRESS_GND, &send_buffer[0],
+            sizeof(send_buffer)/sizeof(send_buffer[0]), MSS_I2C_RELEASE_BUS);
+
+    // Wait for completion and record the outcome
+    status = MSS_I2C_wait_complete(&g_mss_i2c0, MSS_I2C_NO_TIMEOUT);
+
+    if (status == MSS_I2C_SUCCESS)
+    {
+        err = HK_ADC_NO_ERR;
+        return err;
+    }
+
+    err = HK_ADC_ERR_WRITE_FAILED;
+    return err;
+}
+
+
+
+
+/**
+ * @brief HK ADC register read function
+ *
+ * This function performs register read operation. Usually, the value to be
+ * read is a 16-bit value as all the registers accessible for a read
+ * operation in HK ADC are 16-bit registers.
+ *
+ * @param reg[in]              Variable of enumeration type #hk_adc_register_t
+ *                             (defined in hk_adc.h) indicating which register
+ *                             is to be read.
+ * @param read_buffer[in]      pointer to a uint16_t variable to store the result
+ *                             in.
+ * @return HK_ADC_ERR_READ_FAILED if any error occurs otherwise HK_ADC_NO_ERR.
+ *
+ */
+int hk_adc_reg_read(hk_adc_register_t reg, uint16_t *read_buffer)
```

Comment:
Missing a `static` here (for clarity, it would probably not be an issue globally).

Reply:
Yeah, sorry. Added now.